common-libs/internal-api-clients/src/helpers/DidAuthManager.ts

```diff
@@ -2,10 +2,13 @@ import { DidAuthAdapter } from '../helpers/DidAuthAdapter'
 
 export const createDidAuthSession = (didAuthAdapter: DidAuthAdapter) => {
   let responseToken: string | undefined
+  let tokenRequestTime!: number
 
   return {
     async getResponseToken(createRequestToken: (did: string) => Promise<string>) {
-      if (!responseToken || didAuthAdapter.isTokenExpired(responseToken)) {
+      if (!responseToken || didAuthAdapter.isTokenExpired(responseToken, tokenRequestTime)) {
+        tokenRequestTime = Date.now()
```

Comment:
Or alternatively:

* Store `let responseTokenInfo: { tokenRequestTime: string; responseToken: string } | undefined`
* To determine whether you should request new token: check that responseTokenInfo is non-empty or that token is not expired
* `const tokenRequestTime = Date.now()`
* `const responseToken = await didAuthAdapter.createDidAuthResponseToken(...)`
* and in the end, `responseTokenInfo = { tokenRequestTime, responseToken }`

This way, you'll only have one nullable variable; and both you and TS will know that if it's not empty, then both token and request time are non-empty.
Right now, from TS standpoint (and other developers', if they're not reading the implementation really carefully) it is possible that `responseToken` is non-empty yet `tokenRequestTime` is empty (which is probably why you had to declare it with `!` which is a way of saying that you guarantee it will never be empty; but this is wrong: it will be empty prior to the first request to `getResponseToken`)